src/Imath/ImathForward.h

```diff
@@ -17,26 +17,61 @@ IMATH_INTERNAL_NAMESPACE_HEADER_ENTER
 // Basic template type declarations.
 //
 
+//
+// Note: declarations with attributes generate warnings with
+// -Wattributes or -Wall if the type is already defined, i.e. the
+// header is already included. To avoid the warning, only make the
+// forward declaration if the header has not yet been included.
+//
+
+#ifndef INCLUDED_IMATHBOX_H
```

Comment:
You preferred this to adding an include of ImathForward.h to ImathBox.h?

Reply:
This is admittedly kind of ugly, but including ImathForward.h in every .h seems counter to IWYU, and counter-intuitive that including ImathBox.h would give you forward declarations for, say, Quat.